src/main/kotlin/com/team4099/robot2020/subsystems/Shooter.kt

```diff
@@ -0,0 +1,88 @@
+import com.revrobotics.ControlType
+import com.team4099.lib.logging.HelixLogger
+import com.team4099.lib.motorcontroller.SparkMaxControllerFactory
+import com.team4099.lib.subsystem.Subsystem
+import com.team4099.robot2020.config.Constants
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
+import java.lang.Math.abs
+
+object Shooter : Subsystem {
+    private val masterSparkMax = SparkMaxControllerFactory.createDefaultSparkMax(
+            Constants.Shooter.MASTER_SPARKMAX_ID)
+    private val slaveSparkMax = SparkMaxControllerFactory.createPermanentSlaveSparkMax(
+            Constants.Shooter.SLAVE_SPARKMAX_ID, masterSparkMax)
+
+    enum class State {
+        SHOOTING, IDLE, ACCELERATING
+    }
+
+    private var currentSpeed = 0.0
+    var shooterState = State.IDLE
+
+    init {
+        masterSparkMax.pidController.setP(Constants.Shooter.SHOOTER_PID.kP)
+        masterSparkMax.pidController.setI(Constants.Shooter.SHOOTER_PID.kI)
+        masterSparkMax.pidController.setD(Constants.Shooter.SHOOTER_PID.kD)
+        // masterSparkMax.pidController.setFF(Constants.Shooter.SHOOTER_PID.kF)
```

Comment:
can likely uncomment

Reply:
can uncomment after merging develop back into this